DuckDuckGo/MainViewController.swift

```diff
@@ -1774,4 +1788,49 @@ extension MainViewController: VoiceSearchViewControllerDelegate {
     }
 }
 
+
+// MARK: - History UIMenu Methods
+
+extension MainViewController {
+
+    private func refreshBackForwardMenuItems() {
+        guard let currentTab = currentTab, #available(iOS 14.0, *) else {
+            return
+        }
+        omniBar.backButton.menu = historyMenu(with: currentTab.webView.backForwardList.backList.reversed())
+        backButton.menu = historyMenu(with: currentTab.webView.backForwardList.backList.reversed())
+        
+        omniBar.forwardButton.menu = historyMenu(with: currentTab.webView.backForwardList.forwardList)
+        forwardButton.menu = historyMenu(with: currentTab.webView.backForwardList.forwardList)
+    }
+
+    @available(iOS 14.0, *)
+    private func historyMenu(with backForwardList: [WKBackForwardListItem]) -> UIMenu {
+        let historyItemList = backForwardList.map { BackForwardMenuHistoryItem(backForwardItem: $0) }
+        let actions = historyMenuButton(with: historyItemList)
+        return UIMenu(title: "", children: actions)
+    }
+    
+    @available(iOS 14.0, *)
+    private func historyMenuButton(with menuHistoryItemList: [BackForwardMenuHistoryItem]) -> [UIAction] {
+        let menuItems: [UIAction] = menuHistoryItemList.compactMap { historyItem in
+            
+            if #available(iOS 15.0, *) {
+                return UIAction(title: historyItem.title,
+                                subtitle: historyItem.sanitizedURLForDisplay,
+                                discoverabilityTitle: historyItem.sanitizedURLForDisplay) { [weak self] _ in
+                    self?.loadBackForwardItem(historyItem.backForwardItem)
+                }
+            } else {
+                return  UIAction(title: historyItem.title,
+                                 discoverabilityTitle: historyItem.sanitizedURLForDisplay) { [weak self] _ in
+                    self?.loadBackForwardItem(historyItem.backForwardItem)
+                }
+            }
+        }
+        
+        let maxNumberOfItems = 6
+        return Array(menuItems.suffix(maxNumberOfItems))
```

Comment:
One issue here: in the case of the back button, the list is reversed, so taking the suffix means that after a certain point the list won't be accurate.

I just tested this with a back list history of 9 items, and I see 6 in the menu as expected, but it's starting from the oldest item in the list - the last few pages I visited are being truncated.

Reply:
Same as above, I'll remove the 6 limitation. Thanks :)